### DISCUSSION TITLE
settings.RegisterScheme requestContextSettings unable to work properly

### DISCUSSION THREAD

**Discussion by nildebug** [cefsharp/CefSharp#3800] [General]
If you use requestContextSettings to set the view directory. Then settings.RegisterScheme will not work
`var requestContextSettings = new RequestContextSettings
            {
                CachePath = CachePath,
                PersistSessionCookies = true,
                PersistUserPreferences = true
            };
var requestContext = new RequestContext(requestContextSettings);
            var browser = new ChromiumWebBrowser(url)
            {
                RequestContext = requestContext,
                Dock = DockStyle.Fill
            };
`

After using the above code, requestContextSettings cannot work properly, please ask the gods
`const string cefSharpExampleResourcesFolder = @"./Resources";
            settings.RegisterScheme(new CefCustomScheme {
                SchemeName="abc",
                SchemeHandlerFactory=new FolderSchemeHandlerFactory(
                    rootFolder:cefSharpExampleResourcesFolder,
                    schemeName:"abc",
                    hostName:"abc",
                    defaultPage:"home.html"
                    )
});
Cef.Initialize(.......);`
My catalog files are all valid,There is no problem directly using var browser = new ChromiumWebBrowser(url);

**Comment by amaitland** [comment]
In future please use discussions to ask questions. https://github.com/cefsharp/CefSharp/discussions

For a RequestContext you also need to register the schemehandlerfactory. 

http://cefsharp.github.io/api/92.0.x/html/M_CefSharp_IRequestContext_RegisterSchemeHandlerFactory.htm

**Reply by nildebug** [reply]
thank you very much!
